File: tool/compare_l10n.js
const fs = require('fs');
const path = require('path');

const ROOT = path.resolve(__dirname, '..');
const enFile = path.join(ROOT, 'lib', 'l10n', 'app_en.arb');
const arFile = path.join(ROOT, 'lib', 'l10n', 'app_ar.arb');

function load(fp) {
    return JSON.parse(fs.readFileSync(fp, 'utf8'))
}

const en = load(enFile)
const ar = load(arFile)

function keys(obj) {
    return Object.keys(obj).filter(k => !k.startsWith('@'))
}

const enKeys = new Set(keys(en))
const arKeys = new Set(keys(ar))

const onlyEn = [...enKeys].filter(k => !arKeys.has(k)).sort()
const onlyAr = [...arKeys].filter(k => !enKeys.has(k)).sort()

console.log('Keys present only in EN (missing in AR):')
onlyEn.forEach(k => console.log('-', k))
console.log('\nKeys present only in AR (missing in EN):')
onlyAr.forEach(k => console.log('-', k))
console.log('\nCounts:')
console.log('EN total:', enKeys.size)
console.log('AR total:', arKeys.size)
console.log('Only EN:', onlyEn.length)
console.log('Only AR:', onlyAr.length)

File: tool/compare_l10n.py
import json
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
en_file = ROOT / 'lib' / 'l10n' / 'app_en.arb'
ar_file = ROOT / 'lib' / 'l10n' / 'app_ar.arb'

def load(fp):
    return json.loads(fp.read_text(encoding='utf-8'))

en = load(en_file)
ar = load(ar_file)

def keys(d):
    return set(k for k in d.keys() if not k.startswith('@'))

en_keys = keys(en)
ar_keys = keys(ar)

only_en = sorted(en_keys - ar_keys)
only_ar = sorted(ar_keys - en_keys)

print('Keys present only in EN (missing in AR):')
for k in only_en:
    print('-', k)

print('\nKeys present only in AR (missing in EN):')
for k in only_ar:
    print('-', k)

print('\nCounts:')
print('EN total:', len(en_keys))
print('AR total:', len(ar_keys))
print('Only EN:', len(only_en))
print('Only AR:', len(only_ar))
